arviz/data/inference_data.py

```diff
@@ -324,6 +355,311 @@ def sel(
         else:
             return out
 
+    def isel(
+        self, groups=None, filter_groups=None, inplace=False, **kwargs,
+    ):
+        """Perform an xarray selection on all groups.
+
+        Loops groups to perform Dataset.isel(key=item)
+        for every kwarg if key is a dimension of the dataset.
+        One example could be performing a burn in cut on the InferenceData object
+        or discarding a chain. The selection is performed on all relevant groups (like
+        posterior, prior, sample stats) while non relevant groups like observed data are
+        omitted. See :meth:`xarray.Dataset.isel <xarray:xarray.Dataset.isel>`
+
+        Parameters
+        ----------
+        groups: str or list of str, optional
+            Groups where the selection is to be applied. Can either be group names
+            or metagroup names.
+        filter_groups: {None, "like", "regex"}, optional, default=None
+            If `None` (default), interpret groups as the real group or metagroup names.
+            If "like", interpret groups as substrings of the real group or metagroup names.
+            If "regex", interpret groups as regular expressions on the real group or
+            metagroup names. A la `pandas.filter`.
+        inplace: bool, optional
+            If ``True``, modify the InferenceData object inplace,
+            otherwise, return the modified copy.
+        **kwargs: mapping
+            It must be accepted by Dataset.isel().
```

Comment:
modify to use intersphinx, see https://arviz-devs.github.io/arviz/developer_guide.html#hyperlinks

Reply:
Noticed the line:

    We can therefore link to matplotlib docs on Axes.plot from any docstring using :meth:`matplotlib.axes.Axes.plot`.

is wrong, it should be:

    We can therefore link to matplotlib docs on Axes.plot from any docstring using :meth:`mpl:matplotlib.axes.Axes.plot`.

Reply:
For the record, I sent a PR yesterday to fix this, everything should be correct now. There is surely plenty of room for improvement though